test/built-ins/Temporal/PlainDateTime/prototype/withPlainDate/non-compatible-calendars-throw.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.withplaindate
+description: If two non-ISO calendars are involved, an error is raised
+features: [Temporal]
+---*/
+
+const dt = new Temporal.PlainDateTime(1995, 12, 7, 3, 24, 30);
+
+assert.throws(
+  RangeError,
+  () => dt.withCalendar("gregory").withPlainDate("2008-09-06[u-ca=japanese]"),
```

Comment:
This will fail on non-ECMA-402 hosts, so it should use a custom calendar.

Reply:
Could we keep it as-is, but move it to the `intl/` directory?

Reply:
If we did that, the functionality (non-compatible calendars in withPlainDate must throw) wouldn't get tested at all on non-ECMA-402 hosts.

Reply:
fixed by using a fake calendar